lib/tpa/transmogrifiers/replace_2q_repositories.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# © Copyright EnterpriseDB UK Limited 2015-2023 - All rights reserved.
+
+
+from ..transmogrifier import Transmogrifier, opt
+from ..changedescription import ChangeDescription
+from ..checkresult import CheckResult
+
+
+class Replace2qRepositories(Transmogrifier):
+    def __init__(self):
+        # this dict matches the one for new clusters in lib/tpaexec/architectures.py
+        self.postgres_repos = {
+            "postgresql": ["standard"],
+            "edbpge": ["standard"],
+            "pgextended": ["standard", "postgres_extended"],
+            "epas": ["enterprise"],
+        }
+
+    @classmethod
+    def options(cls):
+        return {
+            **opt(
+                "--replace-2q-repositories",
+                action="store_true",
+                help="Replace 2q repositories in config.yml with EDB repositories",
+            )
+        }
+
+    def is_applicable(self, cluster):
+        return True
+        if "edb_repositories" in cluster.vars:
+            return False
+
+        if cluster.vars.get("tpa_2q_repositories") == []:
+            # this means no 2q repos at all, even those added at deploy-time
+            return False
+
+        if cluster.vars.get("postgres_flavour") not in self.postgres_repos:
+            return False
+
+        # if you have been using an explicit yum or apt repo list that doesn't include
+        # PGDG, and you're not going to end up with standard or enterprise, we'd
+        # better not break your cluster
+
+        repository_list = [
+            r
+            for k in cluster.vars
+            if k.endswith("_repository_list")
+            for r in cluster.vars.get(k)
+        ]
+        if (
+            repository_list
+            and "PGDG" not in repository_list
+            and not (
+                self._using_closed_2q_repos(cluster) or postgres_flavour != "postgresql"
+            )
+        ):
+
+            return False
+
+    def check(self, cluster):
+        cr = CheckResult()
+
+        if "tpa_2q_repositories" not in cluster.vars:
+            cr.warning("No 2q repositories present in cluster")
+
+        bdr_version = cluster.vars.get("bdr_version")
+        if bdr_version is not None and bdr_version not in ("3", "4"):
+            cr.error(f"BDR version { bdr_version } cannot be upgraded (must be 3 or 4)")
+
+        return cr
+
+    def _using_closed_2q_repos(self, cluster):
+        tpa_2q_repos = cluster.vars.get("tpa_2q_repositories")
```

Comment:
I think you can save `if tpa_2q_repos is None` conditional in line 77 and 78 by adding a default in previous .get() like:
`cluster.vars.get("tpa_2q_repositories", [])`